src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -19,6 +19,13 @@ function rrule(::typeof(diag), A::AbstractMatrix)
     return diag(A), diag_pullback
 end
 
+function rrule(::typeof(*), D::Diagonal{<:Real}, V::AbstractVector{<:Real})
+    function times_pullback(Ȳ)
+        return (NO_FIELDS, @thunk(Diagonal(Ȳ .* V)), @thunk(D.diag .* Ȳ))
```

Comment:
Do you think it is better to use `parent(D)` rather than accessing a the field directly?
(My benchmark shows they perform identically).
I could go either way.

I would suggest `diag(D)` but that does a copy.
 
```suggestion
        return (NO_FIELDS, @thunk(Diagonal(Ȳ .* V)), @thunk(parent(D) .* Ȳ))
```